sparkle/platform/output.py

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python3
+"""Sparkle class to organise output."""
+
+from __future__ import annotations
+
+from CLI.help import global_variables as gv
+# from sparkle.platform import generate_report_for_selection as sgfs
+from sparkle.platform import \
+    generate_report_for_configuration as sgrfch
+from sparkle.platform.settings_objects import Settings
+from sparkle.solver import Solver
+from sparkle.instance import InstanceSet
+from sparkle.configurator.configuration_scenario import ConfigurationScenario
+from sparkle.configurator.configurator import Configurator
+from sparkle.solver.validator import Validator
+from sparkle.types.objective import SparkleObjective
+
+from CLI.help.nicknames import resolve_object_name
+
+import os
+import json
+from pathlib import Path
+from runrunner.base import Status
+
+
+class ConfigurationPerformance:
+    """Class that stores performance results."""
+    def __init__(self: ConfigurationPerformance, configured_par10: float,
+                 default_par10: float) -> None:
+        """Initalize ConfigurationPerformance.
+
+        Args:
+            configured_par10: The PAR10 result of the configured solver
+            default_par10: The PAR10 result of the default solver
+        """
+        self.configured_PAR10 = configured_par10
+        self.default_PAR10 = default_par10
+
+
+class ValResults:
+    """Class that stores validation information and results."""
+    def __init__(self: ValResults, solver: str, configuration: str, instance_set: str,
+                 results: list[list[str, Status, float, float]]) -> None:
+        """Initalize ValResults.
+
+        Args:
+            solver: The name of the solver
+            configuration: The configuration being used
+            instance_set: The set of instances
+            results: Validation results in the format:
+                [["instance", "status", "quality", "runtime"]]
+        """
+        self.solver = solver
+        self.configuration = configuration
```

Comment:
Configuration can be represented as a dictionary which is cleaner and more useful when you want to port the results to a JSON file